magpie/api/management/resource/resource_views.py

```diff
@@ -212,3 +215,102 @@ def get_res_types(res):
     }
     return ax.valid_http(http_success=HTTPOk, content=data,
                          detail=s.ResourceTypes_GET_OkResponseSchema.description)
+
+
+@s.PermissionsAPI.patch(schema=s.Permissions_PATCH_RequestSchema, tags=[s.PermissionTag],
+                        response_schema=s.Permissions_PATCH_responses)
+@view_config(route_name=s.PermissionsAPI.name, request_method="PATCH")
+def update_permissions(request):
+    """
+    Update the requested permissions and create missing related resources if necessary.
+    """
+    permissions = ar.get_value_multiformat_body_checked(request, "permissions", check_type=list)
+    ax.verify_param(permissions, not_none=True, not_empty=True, http_error=HTTPBadRequest,
+                    msg_on_fail="No permissions to update (empty `permissions` parameter.)")
+
+    required_users = set()
+    required_groups = set()
+    has_permission_to_update = False
+    for entry in permissions:
+        ax.verify_param(entry, is_type=True, param_compare=dict, http_error=HTTPBadRequest,
+                        msg_on_fail="Permission entry should be of `dict` type, but type `{}` was found instead".format(
+                            type(entry)),
+                        content={"param_content": entry})
```

Comment:
For this and following `verify_param`, use `param_content` (function parameter) with the entry provided as `{"value": entry}` instead of `content`. Using `content` overrides other error details added by the response handler.
You can also add the `param_name` argument for cases that help make it more explicit, like when `user`, `group`, `type`, etc. are checked. This will populate the `param` field in the response body.